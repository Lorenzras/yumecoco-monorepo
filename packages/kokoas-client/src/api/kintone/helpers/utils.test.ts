
const form = {
  a : { valueError: true, nextClick: true, page: 1 },
  b : { valueError: false, nextClick: true, page: 1 },
  c : { valueError: false, nextClick: true, page: 1 },
  d : { valueError: true, nextClick: true, page: 1 },
  e : { valueError: true, nextClick: true, page: 2 },
  f : { valueError: true, nextClick: true, page: 2 },
};

test('testlang',
  () => {

    const fieldNamesWithError = Object
      .entries(form)
      .reduce((accu, [key, { valueError, nextClick, page }]) => {
        if (valueError &&  nextClick && page === 1){
          return accu.concat(key);
        }
        return accu;
      }, [] as string[]);



    expect(fieldNamesWithError).toMatchSnapshot();
  },
);