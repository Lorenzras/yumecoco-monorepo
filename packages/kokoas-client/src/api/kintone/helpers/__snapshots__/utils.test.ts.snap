// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testlang 1`] = `
Array [
  "a",
  "d",
]
`;
